### PR TITLE
Update to node 20

### PR DESCRIPTION
**Pull request by tarzacodes**

### CODE REVIEW

**Comment by tarzacodes** [comment]
Most of the changes are due to linting - the linter got updated

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-nodejs 16.15.1
+nodejs 20.11.0
```
Comment (albertobarrila) [inline .tool-versions:1]: the last LTS is 20.15.0

**Review by albertobarrila — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [16.x]
+        node-version: [20.x]
```
Comment (gpad) [inline .github/workflows/main.yml:18]: Why don't keep a matrix like, [16.x, 18.x, 20.x] ?

File: example/tsconfig.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  // "extends": "@tsconfig/node16/tsconfig.json",
+  // "extends": "@tsconfig/node-lts/tsconfig.json",
```
Comment (gpad) [inline example/tsconfig.json:2]: Why is this commented?

File: package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
     "build": "tsc",
     "check": "npm run check-ts && npm run check-lint && npm run check-format && npm run check-spell",
     "check-ts": "tsc --noEmit && tsc --project test/tsconfig.json --noEmit",
-    "check-lint": "tsc --noEmit && eslint 'src/**/*.ts' 'test/**/*.ts'",
+    "check-lint": "tsc --noEmit && eslint --fix 'src/**/*.ts' 'test/**/*.ts'",
```
Comment (gpad) [inline package.json:11]: The fix change the file it's a check so the --fix should not be added

File: package.json [hunk 5]
```diff
@@ -28,30 +28,30 @@
   },
   "homepage": "https://github.com/coders51/rabbitmq-stream-js-client#readme",
   "devDependencies": {
-    "@tsconfig/node16": "^1.0.3",
+    "@tsconfig/node-lts": "^20.1.1",
```
Comment (gpad) [inline package.json:31]: Isn't it deprecated?

File: performance_test/tsconfig.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  // "extends": "@tsconfig/node16/tsconfig.json",
+  // "extends": "@tsconfig/node-lts/tsconfig.json",
```
Comment (gpad) [inline performance_test/tsconfig.json:2]: As above

File: src/amqp10/encoder.ts [hunk 7]
```diff
@@ -47,7 +47,7 @@ type MessageAnnotationsList = { key: string; value: MessageAnnotationsValue }[]
 
 export function amqpEncode(
   writer: DataWriter,
-  { content, messageProperties, applicationProperties, messageAnnotations }: Message
+  { content, messageProperties, applicationProperties, messageAnnotations }: Message,
```
Comment (gpad) [inline src/amqp10/encoder.ts:50]: If you set a property in the prettierc 
```
  "trailingComma": "es5"
```
You don't have to add this orrible comma and you don't have to change all these files

File: example/tsconfig.json (hunk 3 above)
Comment (tarzacodes) [reply example/tsconfig.json:2]: Dunno, found it like this - I'll clean it up since it doesn't seem to be of use

File: package.json (hunk 5 above)
Comment (tarzacodes) [reply package.json:31]: ...is it? I couldn't find anything on google, do you happen to have some source I can look into?

**Review by albertobarrila — APPROVED** [review]

**Pull request merged**